processing/src/main/java/org/apache/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -198,26 +199,33 @@ private ColumnAnalysis analyzeStringColumn(
 
     Comparable min = null;
     Comparable max = null;
-
-    if (!capabilities.hasBitmapIndexes()) {
-      return ColumnAnalysis.error("string_no_bitmap");
-    }
-
-    final BitmapIndex bitmapIndex = columnHolder.getBitmapIndex();
-    final int cardinality = bitmapIndex.getCardinality();
-
-    if (analyzingSize()) {
-      for (int i = 0; i < cardinality; ++i) {
-        String value = bitmapIndex.getValue(i);
-        if (value != null) {
-          size += StringUtils.estimatedBinaryLengthAsUTF8(value) * bitmapIndex.getBitmap(bitmapIndex.getIndex(value)).size();
+    int cardinality = 0;
```

Comment:
My aesthetic sensibility would be to have this be `final` and add a closer `else { }` that sets it to zero, or something. Minor comment though.

Reply:
Changed to do that, thanks for review :metal: